pulsar-functions/api-java/src/main/java/org/apache/pulsar/functions/api/state/StateValue.java

```diff
@@ -18,21 +18,19 @@
  */
 package org.apache.pulsar.functions.api.state;
 
-import java.nio.ByteBuffer;
-
 public class StateValue {
-    private final ByteBuffer value;
+    private final byte[] value;
     private final Long version;
     private final Boolean isNumber;
 
-    public StateValue(ByteBuffer value, Long version, Boolean isNumber) {
-        this.value = value == null ? null : ByteBuffer.wrap(value.array());
+    public StateValue(byte[] value, Long version, Boolean isNumber) {
+        this.value = value == null ? null : value.clone();
         this.version = version;
         this.isNumber = isNumber;
     }
 
-    public ByteBuffer getValue() {
-        return value == null ? null : ByteBuffer.wrap(value.array());
+    public byte[] getValue() {
+        return this.value == null ? null : this.value.clone();
```

Comment:
is it necessary to clone the returned value? I understand that this could make sense in some cases, but is there a real need to have defensive copies?

Reply:
Just to pass the `EI_EXPOSE_REP` check, is it ok to ignore it?

Reply:
Yes, it's fine to omit the check. You can omit it with `@SuppressFBWarnings({"EI_EXPOSE_REP"})` or by adding an entry to the `src/main/resources/findbugsExclude.xml` file. I think that the latter approach is better so that `SuppressFBWarnings` annotation doesn't leak into client API classes.

Reply:
updated